client/src/endpoint/ledger.rs

```diff
@@ -0,0 +1,151 @@
+use error::Result;
+use std::str::FromStr;
+use stellar_resources::Ledger;
+use super::{Body, EndPoint, Order, Records};
+use http::{Request, Uri};
+
+/// Represents the all ledgers end point for the stellar horizon server. The endpoint
+/// will return all ledgers filtered by a myriad of different query params.
+///
+/// https://www.stellar.org/developers/horizon/reference/endpoints/ledgers-all.html
+///
+/// ## Examples
+///
+/// #### Cursor
+///
+/// Starts the page of results at a given cursor
+///
+/// ```
+/// # use stellar_client::sync::Client;
+/// # use stellar_client::endpoint::AllLedgers;
+/// #
+/// let client      = Client::horizon_test().unwrap();
+/// #
+/// # // grab first page and extract cursor
+/// # let endpoint      = AllLedgers::default().limit(1);
+/// # let first_page    = client.request(endpoint).unwrap();
+/// # let cursor        = first_page.next_cursor();
+/// #
+/// let endpoint    = AllLedgers::default().cursor(cursor);
+/// let records     = client.request(endpoint).unwrap();
+/// #
+/// # assert!(records.records().len() > 0);
+/// # assert_ne!(records.next_cursor(), cursor);
+/// ```
+///
+/// #### Order
+///
+/// Fetches all records in a set order, either ascending or descending.
+///
+/// ```
+/// # use stellar_client::sync::Client;
+/// # use stellar_client::endpoint::{AllLedgers, Order};
+/// #
+/// let client      = Client::horizon_test().unwrap();
+/// let endpoint    = AllLedgers::default().order(Order::Asc);
+/// let records     = client.request(endpoint).unwrap();
+/// #
+/// # assert!(records.records().len() > 0);
+/// ```
+///
+/// #### Limit
+///
+/// Fetches a set number of ledgers at a time
+///
+/// ```
+/// # use stellar_client::sync::Client;
+/// # use stellar_client::endpoint::AllLedgers;
+/// #
+/// let client      = Client::horizon_test().unwrap();
+/// let endpoint    = AllLedgers::default().limit(3);
+/// let records     = client.request(endpoint).unwrap();
+/// #
+/// # assert_eq!(records.records().len(), 3);
+/// ```
+#[derive(Debug, Default)]
+pub struct AllLedgers {
+    // // TODO: Cursor needs to be parseable from the links that come back
```

Comment:
I may have left this comment in the AllAssets but I think it can be removed now 😄